### PR TITLE
docs: Add keycloak PKCE and argo-cd cli method documentation

### PR DESCRIPTION
**Pull request by albundy83**
* Add doc to configure keycloak and argo-cd with PKCE, the only way to be able to authenticate to argo-cd with keycloak using web ui AND argo-cd command line.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [x] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [x] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [x] Optional. My organization is added to USERS.md.
* [x] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

Fixes [ISSUE #12124]
Fixes [ISSUE #10029]

### CODE REVIEW

**Comment by albundy83** [comment]
To understand the need on this issue, I could mention this issue https://github.com/argoproj/argo-cd/issues/12124 and also this one https://github.com/argoproj/argo-cd/issues/10029.
My aim is to simplfy the way to correctly configure keycloak when you want to authenticate also using cli.

File: USERS.md [hunk 1]
```diff
@@ -95,6 +95,7 @@ Currently, the following organizations are **officially** using Argo CD:
 1. [Devtron Labs](https://github.com/devtron-labs/devtron)
 1. [DigitalOcean](https://www.digitalocean.com)
 1. [Divistant](https://divistant.com)
+1. [Docaposte](https://docaposte.com)
```
Comment (andrii-korotkov-verkada) [inline USERS.md:98]: Let's put this in a separate PR.

File: docs/operator-manual/user-management/keycloak.md [hunk 2]
```diff
@@ -19,62 +28,97 @@ Enable the __Client authentication__.
...
+Leave default values.
+
+![Keycloak add client Step 2](../../assets/keycloak-add-client-pkce_2.png "Keycloak add client Step 2")
+
+Configure the client by setting the __Root URL__, __Web origins__, __Admin URL__ to the hostname (https://{hostname}).
+
+Also you can set __Home URL__ to _/applications_ path and __Valid Post logout redirect URIs__ to "https://{hostname}/applications".
+
+The Valid Redirect URIs should be set to:
+- http://localhost:8085/auth/callback (needed for argo-cd cli)
```
Comment (andrii-korotkov-verkada) [inline docs/operator-manual/user-management/keycloak.md:108]: Is localhost here correct, i.e. not just used for testing?

File: docs/operator-manual/user-management/keycloak.md [hunk 3]
```diff
@@ -91,19 +135,53 @@ data:
...
-    clientSecret: $oidc.keycloak.clientSecret
+    enablePKCEAuthentication: true
     requestedScopes: ["openid", "profile", "email", "groups"]
 ```
 
 Make sure that:
 
 - __issuer__ ends with the correct realm (in this example _master_)
-- __issuer__ on Keycloak releases older than version 17 the URL must include /auth (in this example /auth/realms/master)
+- __issuer__ on Keycloak releases older than version 17 the URL must includenablePKCEAuthentication: truee /auth (in this example /auth/realms/master)
```
Comment (andrii-korotkov-verkada) [inline docs/operator-manual/user-management/keycloak.md:145]: Typos: includenablePKCEAuthentication, truee

File: USERS.md (hunk 1 above)
Comment (albundy83) [reply USERS.md]: I have removed it

File: docs/operator-manual/user-management/keycloak.md (hunk 3 above)
Comment (albundy83) [reply docs/operator-manual/user-management/keycloak.md]: Sorry by copy / paste

File: docs/operator-manual/user-management/keycloak.md (hunk 2 above)
Comment (albundy83) [reply docs/operator-manual/user-management/keycloak.md]: Yes, as when you try to authenticate using sso, argo-cd cli starts a listener on localhost with default port 8085.
It also launcher a web browser where you will fill your login and password.
Then, the callback send your token to argo-cd cli listener.
That's why http://localhost:8085/auth/callback must be allowed.

**Review by andrii-korotkov-verkada — APPROVED** [review]

File: docs/operator-manual/user-management/keycloak.md [hunk 4]
```diff
@@ -19,62 +28,97 @@ Enable the __Client authentication__.
...
+### Configuring ArgoCD OIDC
 
-In order for ArgoCD to provide the groups the user is in we need to configure a groups claim that can be included in the authentication token.
-To do this we'll start by creating a new __Client Scope__ called _groups_.
+Let's start by storing the client secret you generated earlier in the argocd secret _argocd-secret_.
 
-![Keycloak add scope](../../assets/keycloak-add-scope.png "Keycloak add scope")
+You can patch it with value copied previously:
+```bash
+kubectl -n argo-cd patch secret argocd-secret --patch='{"stringData": { "oidc.keycloak.clientSecret": "83083958-8ec6-47b0-a411-a8c55381fbd2" }}'
```
Comment (ishitasequeira) [inline docs/operator-manual/user-management/keycloak.md:50]: nit: Instead of passing the random string value, we can maybe set it to `<REPLACE_WITH_CLIENT_SECRET>`. WDYT?

**Review by ishitasequeira — APPROVED** [review]
@albundy83, other than the nit above, the PR looks good to me.

File: docs/operator-manual/user-management/keycloak.md (hunk 4 above)
Comment (albundy83) [reply docs/operator-manual/user-management/keycloak.md]: I think you are absolutely right, I have updated my pull request :)

**Review by ishitasequeira — APPROVED** [review]
